### DISCUSSION TITLE
How fix lint operator does not exist: uuid = text

### DISCUSSION THREAD

**Discussion by sergiohdljr** [ts-safeql/safeql#198] [Q&A] [answered]
I have the following code : 

![Captura de tela de 2024-01-13 19-59-45](https://github.com/ts-safeql/safeql/assets/102623806/de3e43a8-1899-4524-b3a5-26617102fae4)

 On this table the column id has a UUID type, wich i am comparing with a id send from the params, that is actually a UUID but as a string. 

any ideias how i could fix that ? (i am open for database remodelation too)

**Comment by pokedotdev** [comment]
this is the way
```ts
sql`DELETE FROM farmacia WHERE id = ${id}::uuid`
```

**Reply by sergiohdljr** [reply]
> this is the way
> ```ts
> sql`DELETE FROM farmacia WHERE id = ${id}::uuid`
> ```


i tried this way and got a sintax error, then try like this and work. but still dont sound the best way its like use 'as' in typescript.

 ```ts
> sql`DELETE FROM farmacia WHERE id::text = ${id}`
> ```

**Reply by pokedotdev** [reply]
the syntax error may be because the UUID you are entering is invalid. On the other hand, I think that casting from uuid to text would give worse performance.

**Reply by Newbie012** [reply]
You could create a new TS type called UUID (a branded type) and tell SafeQL to treat it as UUID via `overrides.types`

**Reply by sergiohdljr** [reply]
not sure how do that, but i will try

**Reply by Newbie012** [reply, marked as answer]
@sergiohdljr there are many libraries which support branded types such as zod's [`brand`](https://github.com/colinhacks/zod?tab=readme-ov-file#brand).

First, define your UUID brand:

```ts
import { z } from "zod";

const UUID = z.string.brand<"UUID">();
type UUID = z.infer<typeof UUID>;
```

Second, tell SafeQL to treat `uuid` as your branded `UUID` (instead of the default `string`) using [`overrides.types`](https://safeql.dev/api/#connections-overrides-types-optional):

```ts
{
  "connections": {
    "overrides": {
      "types": {
        "uuid": "UUID"
      }
    }
  }
}
```

Lastly, this is how you code should like:

```ts
import { UUID } from "./uuid";

function run(id: UUID) {
  return sql`DELETE FROM farmacia WHERE id = ${id}`
}
```
